src/main/java/ch/hftm/controllers/AlbumController.java

```diff
@@ -95,7 +122,60 @@ private void createLabel(GridPane pane, int column, String labelText) {
         }
     }
 
+    @FXML
+    public void deleteImage() {
+        Alert alert = new Alert(AlertType.CONFIRMATION);
+        alert.setTitle("Delete selected images");
+        alert.setContentText("If you want to delete the selected images press OK.");
+        ButtonType okButton = new ButtonType("OK");
+        ButtonType cancelButton = new ButtonType("Cancel");
+        alert.getButtonTypes().setAll(okButton, cancelButton);
+        Optional<ButtonType> choice = alert.showAndWait();
+
+        if (choice.isPresent() && choice.get() == okButton) {
+            ImageService imageService = new ImageService();
+            if (!(imagesToDelete.isEmpty())) {
+                int i = 0;
+                for (Image image : imagesToDelete) {
+                    imageService.deleteImage(image);
+                    i++;
+                }
+                updateAlbumStatus(String.format("Deleted %d image(s).", i), 5_000);
+                initializeAlbum();
+            } else {
+                Alert info = new Alert(AlertType.INFORMATION);
+                info.setTitle("Invalid selection");
+                info.setContentText("No images where selected.");
+                info.showAndWait();
+            }
+        } else {
+            for (Node child : imageGrid.getChildren()) {
+                child.setStyle(null);
+            }
+        }
+        imagesToDelete.clear();
```

Comment:
You could also write the code like this, it should do the same thing, but for me personally is easier to read, since it does not have that much nesting. 
```suggestion
        if (!choice.isPresent() || choice.get() != okButton) {
            for (Node child : imageGrid.getChildren()) {
                child.setStyle(null);
            }
            imagesToDelete.clear();
            return;
        }

        if (imagesToDelete.isEmpty()) {
            Alert info = new Alert(AlertType.INFORMATION);
            info.setTitle("Invalid selection");
            info.setContentText("No images where selected.");
            info.showAndWait();
            imagesToDelete.clear();
            return;
        }

        ImageService imageService = new ImageService();
        int i = 0;
        for (Image image : imagesToDelete) {
            imageService.deleteImage(image);
            i++;
        }
        updateAlbumStatus(String.format("Deleted %d image(s).", i), 5_000);
        initializeAlbum();
```

Reply:
I refactored it because it definitly is more readable.